### PR TITLE
Merge properties from vector tiles into feature info

### PR DESCRIPTION
**Pull request by tephenavies**
### What this PR does

Re-implement vector tile feature & csv row property merging in v8.

### Checklist

-   [x] Not easily testable (best I can think of is: set a custom regionMapping, have a small sample vector tileset with just a 0/0/0 tile, have a small sample csv that uses the tileset, emulate feature picking)
-  [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

File: lib/ModelMixins/TableMixin.ts [hunk 1]
```diff
@@ -717,7 +717,7 @@ function TableMixin<T extends Constructor<Model<TableTraits>>>(Base: T) {
 
                 return this.featureInfoFromFeature(
                   regionType,
-                  d,
+                  Object.assign({}, feature.properties, d),
```
Comment (nf-s) [inline lib/ModelMixins/TableMixin.ts]:
```suggestion
                  Object.assign(
                    {},
                    d,
                    // Only add feature.properties which don't exist in $d
                    Object.entries(feature.properties).reduce<any>(
                      (props, [key, value]) => {
                        if (d![key] === undefined) {
                          props[key] = value;
                        }
                        return props;
                      },
                      {}
                    )
                  ),
```

Is it worth reversing the order of properties so column values are listed above region feature properties in FeatureInfoSection?

**Review by nf-s — CHANGES_REQUESTED** [review]

File: lib/ModelMixins/TableMixin.ts (hunk 1 above)
Comment (tephenavies) [reply lib/ModelMixins/TableMixin.ts]: Good catch. I raise your suggestion with:
```suggestion
                  Object.assign(d, Object.assign({}, feature.properties, d)),
```

I know this modifies d, but we were modifying it previously in `featureInfoFunc`. Or:
```suggestion
                  Object.assign({}, d, Object.assign({}, feature.properties, d)),
```
Or we could use what you've written if you think that conveys intent better.

------

(the same as above but with a quick comment, but maybe not a good comment)
```suggestion
                  // Preserve values from d and insert feature properties after entries from d
                  Object.assign(d, Object.assign({}, feature.properties, d)),
```

Not modifying d:
```suggestion
                  // Preserve values from d and insert feature properties after entries from d
                  Object.assign({}, d, Object.assign({}, feature.properties, d)),
```

Comment (nf-s) [reply lib/ModelMixins/TableMixin.ts]: Yep the last one looks good to me!
But is it equivalent to `Object.assign({}, d, feature.properties, d)`?

image

Comment (nf-s) [reply lib/ModelMixins/TableMixin.ts]: > Properties in the target object are overwritten by properties in the sources if they have the same key. Later sources' properties overwrite earlier ones.

Comment (tephenavies) [reply lib/ModelMixins/TableMixin.ts]: Yep. That's equivalent

Comment (tephenavies) [reply lib/ModelMixins/TableMixin.ts]:
```suggestion
                  // Preserve values from d and insert feature properties after entries from d
                  Object.assign({}, d, feature.properties, d),
```

**Review by nf-s — APPROVED** [review]
soz forgot to merge

**Pull request merged**